runtime/context.go

```diff
@@ -124,6 +201,32 @@ func accumulateLogFields(ctx context.Context, newFields []zap.Field) []zap.Field
 	return fields
 }
 
+// ContextExtractor is a extractor that extracts some log fields from the context
+type ContextExtractor interface {
+	ExtractScope(ctx context.Context, key string) map[string]string
+}
+
+// NewContextExtractor returns a extractor that extracts log fields a context.
+func NewContextExtractor() ContextExtractor {
+	return &contextExtractor{}
+}
+
+type contextExtractor struct {
+}
+
+// ExtractScope extracts scope fields from a context into a tag.
+func (c *contextExtractor) ExtractScope(ctx context.Context, key string) map[string]string {
```

Comment:
I figured all extractors would be used at every call site regardless of whether it's a client or an endpoint. For example, if the user defines a custom type `task` that defines their own context fields, the client metrics in zanzibar should pick up those task fields. As written this would require the user to carefully configure the map they return in the App struct as:

```
var App = &zanzibar.App{
    GetContextScopeExtractors: func() map[string]ContextExtractor {
        return map[string]ContextExtractor{
            "client": GetTaskAndClientTags,
            "task": GetTaskTags,
        }
    },
}
```

That is, they would need to be careful that the extractor for `"client"` returns task tags too, else the client metrics in zanzibar will not pick up the task tags. We don't really need that separation though, instead we can just return a list of functions that's run every time. While it does waste a bit of computation, it's probably a bit easier to understand:

```
var App = &zanzibar.App{
    GetContextScopeExtractors: func() []ContextExtractor {
        return []ContextExtractor{
            GetClientTags,
            GetTaskTags,
            GetRegionHeaderTags,
        }
    },
}
```